### PR TITLE
Overclocked GPU/BUS,added shaders,disabled vsync.

### PR DESCRIPTION
**Pull request by Rinnegatamante**
This pull will do four different things:

- Adds filters (aka shaders) support (at the moment there are 4 shaders available: None, Sharp Bilinear, LCD 3x, xBR). These will be interchangeable during player execution by the user by swinging the touchscreen left/right.

- Increases GPU, GPU crossbar and BUS clocks (from 111 mhz to 222 mhz) allowing shaders to get the best results.

- Disables vsync. This will produce a slight frameboost without any bad conseguence.

- Fix some typo errors and updates Makefile for new vitasdk compatibility.

### CODE REVIEW

**Comment by Ghabry** [comment]
You messed up the rebase, needs a squash.

File: src/psp2_ui.h [hunk 1]
```diff
@@ -27,13 +27,9 @@
...
-vita2d_shader* shaders[SHADERS_NUM];
-char* shader_names[] = {
-	"None",
-	"Sharp Bilinear",
-	"LCD 3x",
-	"xBR"
-};
+
+extern vita2d_shader* shaders[];
+extern char* shader_names[];
```
Comment (Ghabry) [inline src/psp2_ui.h]: Why are you declaring the shaders as extern? Are they used anywhere outside of psp2_ui.cpp?

File: src/psp2_ui.cpp [hunk 2]
```diff
@@ -47,14 +58,30 @@ AudioInterface& Psp2Ui::GetAudio() {
 
 int _newlib_heap_size_user = 192 * 1024 * 1024;
 
+vita2d_shader* shaders[SHADERS_NUM];
+char* shader_names[] = {
+	"None",
+	"Sharp Bilinear",
+	"LCD 3x",
+	"xBR"
+};
```
Comment (Ghabry) [inline src/psp2_ui.cpp]: If the reason for extern are strange linker errors: Declare these variables as "static"

**Comment by Ghabry** [comment]
Just noticed that the Makefile lacks a check if TOOLCHAIN_DIR is defined, but isn't your fault.

**Comment by Ghabry** [comment]
I get linker errors.
Please move the global variables above class Psp2Ui from psp2_ui.h into an anonymous namespace (namespace {}) in the .cpp file. Fixes it for me

**Comment by Ghabry** [comment]
A user on gbatemp reported that this made (even with shaders disabled) some games, e.g. Ara Fell, significantly slower. Theres no obvious reason why. Could be related to how the upstream-version handles audio, lets wait for the new audio code to be finished.

**Comment by Ghabry** [comment]
@Rinnegatamante 
A user reports a massive slowdown (15 FPS) in "Ara Fell" since the gpu-code changes, even in my "sdl_con_audio_sin_mixer" branch. Maybe scheduling or new vitasdk problem.

Ara Fell Demo Download:
http://queenscourt.org/othergames.php

**Comment by Rinnegatamante** [comment]
I doubt it's an issue related to vitasdk, maybe it could be related to the new gpu rendering thread + lots of sounds thread working.

**Comment by Ghabry** [comment]
We confirmed already that even with the new audio code it runs bad, but havn't tested rolling back to the non-render-thread version yet.

**Pull request merged**